javalin/src/main/java/io/javalin/http/servlet/JavalinServletRequest.kt

```diff
@@ -0,0 +1,13 @@
+package io.javalin.http.servlet
+
+import jakarta.servlet.ServletInputStream
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletRequestWrapper
+
+class JavalinServletRequest(request: HttpServletRequest) : HttpServletRequestWrapper(request) {
+    override fun getInputStream(): ServletInputStream =
+        super.getInputStream().also { inputStreamRead = true }
+
+    private var inputStreamRead = false
+    fun isInputStreamRead() = inputStreamRead
```

Comment:
I'd mark this class as internal and put the read variable on top as a property with private set:

```kotlin
var inputStreamRead: Boolean = false
    private set
```

Also, this is technically not true that it's read if accessed via getInputStream - a more accurate solution would need to listen for InputStream's method calls. Hopefully it's good enough

Reply:
I know, but that sounds a bit overkill. This is cheap and should cover 99%.

Reply:
Will fix property, this is much better!